packages/sdk/src/e2e_reliability/index.spec.ts

```diff
@@ -0,0 +1,282 @@
+  }
+  public isConnected(): boolean {
+    throw new Error("Method not implemented.");
+  }
+  public getConnectedPeers(): Promise<Peer[]> {
+    throw new Error("Method not implemented.");
+  }
+}
+
+describe("E2E Reliability", () => {
+  let mockWakuNode: IWaku;
+  let encoder: IEncoder;
+  let decoder: IDecoder<IDecodedMessage>;
+
+  beforeEach(async () => {
+    mockWakuNode = new MockWakuNode();
+    encoder = createEncoder({
+      contentTopic: TEST_CONTENT_TOPIC,
+      routingInfo: TEST_ROUTING_INFO
+    });
+    decoder = createDecoder(TEST_CONTENT_TOPIC, TEST_ROUTING_INFO);
+  });
+
+  it("Outgoing message is emitted as sending", async () => {
+    const messageChannel = MessageChannel.create(mockWakuNode, "MyChannel");
+
+    const subRes = await messageChannel.subscribe(decoder);
+    expect(subRes).to.be.true;
+
+    const message = { payload: utf8ToBytes("message in channel") };
+
+    // Setting up message tracking
+    const messageId = MessageChannel.getMessageId(message.payload);
+    let messageSending = false;
+    messageChannel.addEventListener(
+      MessageChannelEvent.OutMessageSending,
+      (event) => {
+        if (event.detail === messageId) {
+          messageSending = true;
+        }
+      }
+    );
+
+    await messageChannel.send(encoder, message);
+
+    expect(messageSending).to.be.true;
+  });
+
+  it("Outgoing message is emitted as sent", async () => {
+    const messageChannel = MessageChannel.create(mockWakuNode, "MyChannel");
+
+    const subRes = await messageChannel.subscribe(decoder);
+    expect(subRes).to.be.true;
+
+    const message = { payload: utf8ToBytes("message in channel") };
+
+    // Setting up message tracking
+    const messageId = MessageChannel.getMessageId(message.payload);
+    let messageSent = false;
+    messageChannel.addEventListener(
+      MessageChannelEvent.OutMessageSent,
+      (event) => {
+        if (event.detail === messageId) {
+          messageSent = true;
+        }
+      }
+    );
+
+    await messageChannel.send(encoder, message);
+
+    expect(messageSent).to.be.true;
+  });
+
+  it("Outgoing message is emitted as acknowledged", async () => {
+    const messageChannel = MessageChannel.create(mockWakuNode, "MyChannel");
+
+    const subRes = await messageChannel.subscribe(decoder);
+    expect(subRes).to.be.true;
+
+    const message = { payload: utf8ToBytes("first message in channel") };
+
+    // Setting up message tracking
+    const messageId = MessageChannel.getMessageId(message.payload);
+    let messageAcknowledged = false;
+    messageChannel.addEventListener(
+      MessageChannelEvent.OutMessageAcknowledged,
+      (event) => {
+        if (event.detail === messageId) {
+          messageAcknowledged = true;
+        }
+      }
+    );
+
+    await messageChannel.send(encoder, message);
+
+    // Sending a second message to acknowledge the first..
+    // TODO: Let's see if we are acknowledging our own messages... we are!
```

Comment:
`MessageChannel` marks `message` as *acknowledged* because it saw the second message sent below.

But they are both from same sender... a message sent locally should not acknowledge other messages sent locally.